src/host/toolsHost.ts

```diff
@@ -80,11 +80,17 @@ export default class ToolsHost {
         encodeMessageForChannel((msg) => window.parent.postMessage(msg, "*"), "openInEditor", request);
     }
 
+    public getApprovedTabs(callback: () => void) {
+        const id = this.getHostCallbacksNextId++;
+        this.getHostCallbacks.set(id, callback);
+        encodeMessageForChannel((msg) => window.parent.postMessage(msg, "*"), "getApprovedTabs", {id});
+    }
+
     public onMessageFromChannel(e: WebviewEvent, args: string): boolean {
         switch (e) {
             case "getState": {

```

Comment:
Could you go into a little more detail into why we want to make the distinction between "getState" and "getHost"?  It appears that we care enough to change the names of the variables in this file, but not enough to create a new webview event.  To me, it just seems odd that now we're firing "fireGetHostCallback" on "getState."

Happy to keep this change, but if we do decide to change it, it should also be reflected in the toolsHost.test.ts file as well in places such as here: https://github.com/microsoft/vscode-edge-devtools/blob/master/src/host/toolsHost.test.ts#L44

If we want to make the distinction between "getState" vs "getHost", do you think we should make the same changes for the

Reply:
Sure, 
getState is a webview event created to retrieve the "state" for the devtools, meaning the preferences that devtools require for initialization. To achieve this it uses a synchronization mechanism:

    private **getStateCallbacks**...

While I was coding the solution for the network I realized that my code was looking pretty similar to this, so I generalized the implementation to avoid code repetition. Thus becoming:

    private **getHostCallbacks:**...

GetState is still an event that hints that devtools preferences need to be retrieved, but it will now use the generalized implementation **fireGetHostCallback** to synchronize. Does this answer the question?

Reply:
Yeah that makes sense.  LGTM then.